### ISSUE TITLE
Display of predicted walking time

### ISSUE THREAD

**Issue by pingurus** [Maproom/qmapshack#35]
**Is your feature request related to a problem? Please describe.**
Sometimes when I plan hikes and I have a specific timeframe it would be nice if QMS displayed the time needed for track.

**Describe the solution you'd like**
[Here](https://de.wikipedia.org/wiki/Marschzeitberechnung) (sorry, only german) you'll find the formula of DIN 33466 according to which the times on the hiking signs in the alps are labelled.

The calculation base is that in one hour you can cover:
* 300m of elevation gain in ascent
* 500m of elevation gain in descent
* 4km distance in flat areas.

First, the times needed for the elevation and distance are calculated seperately. Then the smaller one of the two is added to the bigger one to get the total time.

Example:

- Distance: 23km Elevation gain: 1025m Elevation loss: 1725m
- Time needed for distance: 23km/4km/h = 5,75h
- Time needed for elevation: 1025m/300m/h + 1725m/500m/h = 6,86h
- Nominal time needed: 6,86h+5,75h/2=9,74h

As a comparison we needed 9,15h for that hike, so this is a rather conservative approach, just as said in the wikipedia article.

**Describe alternatives you've considered**
Alternatively one could also use [this](https://de.wikipedia.org/wiki/Leistungskilometer) (again, only german) concept, however I prefer the first one since it outputs times.

**Additional context**
Of course, this calculation should be only triggered if the profile is set to 'foot'

**Comment by kiozen** [comment]
You are aware of the speed filter?

https://github.com/Maproom/qmapshack/wiki/AdvTrkFilters#hiking

**Comment by pingurus** [comment]
No, I was not.
I think we should still keep it, as it does not tamper with the actual track data. This way one can get a feeling how fast one is compared to the signs, as it is also calculated for recorded tracks.
But I understand if you think my use-case is already covered by the filter.

**Comment by pingurus** [comment]
Whilst playing around with the filter, I was thinking wether it would be possible to give the user the possibility to automatically import track data into different profiles. These profiles would average the data fed into them depending on incline and distance until this point to create a profile corresponding to the users capabilities. Then, instead of the formula used by the filter or by the DIN, the predicted time and the filter results could be based on this generated profile.
However, I'm not sure if this is really needed/wanted.
So, do you think this is an overkill?
Do you think it is useful?
Do you think it is feasible in a reasonable amount of time? (My semester starts again in two weeks of time and I probably won't have that much time to code from that point on + I think we'd have to make the activity system work first, so we can connect the generated profiles to activities. Currently there are too few activities to make it work (I'd need at least two different biking profiles for it to be useful))

**Comment by kiozen** [comment]
Sounds quite complex. But if you are interested go ahead. This is a fun project :smile: 

But I agree that enhancing the activity system will have a much broader benefit to users. :+1:

**Comment by kiozen** [comment]
> I think we should still keep it, as it does not tamper with the actual track data. This way one can get a feeling how fast one is compared to the signs, as it is also calculated for recorded tracks.

:+1:

**Comment by mitxel-m** [comment]
Hi all

about the formula:
My first impression with that formula is that the results aren't very realistic.
I will try the PR :-) and compare with the actual filter with the same parameters ( 4Km/h, 300m/h, 500m/h)

about personal profiles:
Deduce a personal profile based on a bunch of tracks would require that all tracks studied were strictly correct (no points or aberrant, no aberrant heights). This means a lot of cleaning work hence a lot of time. Only a couple of points some km away from the starting point in a track would lead to a false profile.

However I see much easier that users **really** interested on getting time predictions  check some sections of real and well known tracks to get their usual values as described in the wiki. You do it only once and since then you can use your custom values in the filter. 

That is also a way  to get the user personally involved in knowing their values when preparing activities.


I also agree that enhancing the activity system will have a much more benefit. Specially doing something that maybe in the future could connect some activities with some time filters. 
I mean apply different time profiles in various ranges of a single track based on the activity of each range.

;-)

**Comment by pingurus** [comment]
When #39 is done, I would suggest having one clearly labelled, editable default filter for each activity. This default filter could then be used to calculate the time diplayed, just as they could be used to do the 'invasive' version of editing the track data.
Let's focus on the activities for now.

**Comment by kiozen** [comment]
As this issue is older than 6 months I close it for now.
